Source/CBL_SQLiteStorage.m

```diff
@@ -677,11 +677,18 @@ - (SInt64) _getDocNumericID: (NSString*)docID {
 - (SInt64) _createDocNumericID: (NSString*)docID {
     if (![_fmdb executeUpdate: @"INSERT OR IGNORE INTO docs (docid) VALUES (?)", docID])
         return -1;
-    if (_fmdb.changes == 0)
-        return -1;
+    if (_fmdb.changes == 0) {
+      SInt64 result = [self _getDocNumericID:docID];
+      if(result > 0) return result; 
+      return -1;
+    }
     return _fmdb.lastInsertRowId;
 }
 
+-(void) removeKeyFromDocIDCache:(NSString*)docID {
```

Comment:
You don't need to add this method. It would be better to have the unit test close and re-open the database, by calling -reopenTestDB.